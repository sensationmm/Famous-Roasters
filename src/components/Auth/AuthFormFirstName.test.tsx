import { MockedProvider } from '@apollo/client/testing'
import { render, waitFor } from '@testing-library/react'
import Form from 'rc-field-form'
import React from 'react'

import { AuthFormFirstName } from '.'

describe('First name fields group custom auth component', () => {
  const snippet = (value: string | undefined = undefined) => (
    <Form name="testAuthFormItemInput" method="POST">
      <AuthFormFirstName screenKey="signIn" onChange={() => null} value={value} />
    </Form>
  )

  it('Renders correctly', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>{snippet()}</MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly with given value', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>{snippet('test')}</MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
