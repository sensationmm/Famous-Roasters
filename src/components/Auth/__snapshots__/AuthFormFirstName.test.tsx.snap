// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First name fields group custom auth component Renders correctly 1`] = `
<div>
  <form
    method="POST"
  >
    <label
      class="relative"
    >
      <span
        class="absolute -top-6 left-3 px-1 bg-white"
      >
        <span
          class="text-sm leading-5"
        >
          auth.signIn.firstName.label
        </span>
      </span>
      <input
        class="w-full border rounded-full px-4 py-2 focus:outline-0 focus:ring focus:ring-primary/30"
        data-testid="firstName"
        placeholder="auth.signIn.firstName.placeholder"
        type="text"
        value=""
      />
    </label>
  </form>
</div>
`;
